cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/DatasetContextLookupProvider.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.etl.common;
+
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.etl.api.Lookup;
+import co.cask.cdap.etl.api.LookupProvider;
+
+import java.util.Map;
+
+/**
+ * Implementation of {@link Lookup} that uses {@link DatasetContext}.
+ */
+public class DatasetContextLookupProvider implements LookupProvider {
+
+  private final DatasetContext context;
+
+  public DatasetContextLookupProvider(DatasetContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public <T> Lookup<T> provide(String table, Map<String, String> arguments) {
+    Dataset dataset = context.getDataset(table, arguments);
+    if (dataset == null) {
+      throw new RuntimeException(String.format("Dataset %s does not exist", table));
+    }
+
+    if (!(dataset instanceof Lookup)) {
```

Comment:
So `Lookup` is in etl-api, right? Meaning no dataset is actually implementing `Lookup`?